docs/authoring.md

```diff
@@ -343,4 +343,21 @@ You can use the filter function in your template as in the following example:
 const {{ channelName | camelCase }} = '{{ channelName }}';
 ```
 
+The generator supports also asynchronous filters. Asynchronous filters receive as last argument a callback to resume rendering. Asynchronous filters must be annotated with the `async` keyword. Make sure to call the callback with two arguments: `callback(err, res)`. `err` can be `null`. See the example of how to use asynchronous filters:
```

Comment:
```suggestion
The generator also supports asynchronous filters. Asynchronous filters receive as last argument a callback to resume rendering. Asynchronous filters must be annotated with the `async` keyword. Make sure to call the callback with two arguments: `callback(err, res)`. `err` can be `null`. See the following example of how to use asynchronous filters:
```

What about filters that return a promise? 🤔

Reply:
Nunjucks doesn't support it. Determine that function returns Promise is impossible without invoke it, so we must think about another way to resolve problem.

Reply:
Yea, let's not focus on that at all, if people want it they will ask for it 😄

Reply:
My first thought how to resolve that: we should return as filter not function but object with metadata that function is asynchronous, like

```js
const filters = module.export;

filters.someFilter = {
  isAsync: true,
  fn: function() { return new Promise() ... }
}
```